### ISSUE TITLE
[Bug]: Broken automatic updates | env: ‘./netdata-installer.sh’: Permission denied

### ISSUE THREAD

**Issue by therealgilles** [netdata/netdata#14126]
### Bug description

Here is what I get in the automatic update log:

```
Fri Dec  9 06:44:59 AM PST 2022 : INFO: netdata-updater:  Re-installing netdata...
env: ‘./netdata-installer.sh’: Permission denied
Fri Dec  9 06:44:59 AM PST 2022 : FATAL: netdata-updater: FAILED TO UPDATE NETDATA:  Failed to rebuild existing netdata reinstall.
run-parts: /etc/cron.daily/netdata-updater exited with return code 1
```

### Expected behavior

Automatic update should work without errors. Proper permissions should be set to run the script.

### Steps to reproduce

Trigger an automatic update.

### Installation method

kickstart.sh

### System info

```shell
Linux ols-server 5.15.0-53-generic #59-Ubuntu SMP Mon Oct 17 18:53:30 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=22.04
/etc/lsb-release:DISTRIB_CODENAME=jammy
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
/etc/os-release:PRETTY_NAME="Ubuntu 22.04.1 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION_ID="22.04"
/etc/os-release:VERSION="22.04.1 LTS (Jammy Jellyfish)"
/etc/os-release:VERSION_CODENAME=jammy
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:UBUNTU_CODENAME=jammy
```

### Netdata build info

```shell
Version: netdata v1.37.1
Configure options:  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--with-bundled-protobuf' 'CFLAGS=-O2 -pipe' 'LDFLAGS='
Install type: legacy-build
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES 
    ACLK:                       YES
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
Libraries:
    protobuf:                YES (bundled)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    YES
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: NO
Debug/Developer Features:
    Trace Allocations:       NO
```

### Additional info

This has been a recurring issue. There ought to be better testing to make sure it does not happen again.

**Comment by Dim-P** [comment]
Closing as there hasn't been a response for over a month and this seems to be a one-off case. We haven't seen any other users reporting this issue. If it still persists after following the above steps, please comment here and we will reinvestigate it.

**Comment by therealgilles** [comment]
Sorry for the late reply. Here are the script permissions:
```
-rwxr-xr-x 1 root root 31045 Feb 15 07:15 /usr/libexec/netdata/netdata-updater.sh
```

Could you tell me how to install native packages on ubuntu?

**Comment by Dim-P** [comment]
HI @therealgilles . How did you install Netdata in the first place (it seems like you installed it from Git)? Did you try to run the updater as root to see if you are still getting the same error?

To install the native packages on Ubuntu, please run: `wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh` as recommended in our [Get started with Netdata](https://learn.netdata.cloud/docs/get-started#install-on-linux-with-one-line-installer) guide. Do not use `apt install netdata` as this will install the Ubuntu repos Netdata version, which we do not officially support. 

I would recommend you [do a complete uninstall](https://learn.netdata.cloud/docs/agent/packaging/installer/uninstall) of Netdata first.

**Comment by therealgilles** [comment]
Hi @Dim-P, I had installed netdata following the instructions, i.e. using the `kickstart.sh` script.

First I'm told to use native ubuntu packages, which to me means using `apt install` and then you tell me I should not. This is really confusing.

This permission problem has been a recurring problem for me. I thought it was fixed at one point and then it re-appeared without my doing anything, so I'm pretty sure it's a problem on your side.

I will try the full uninstall / re-install.

**Comment by Dim-P** [comment]
Hi @therealgilles . Let me explain that. We maintain our own repositories that get installed via the `kickstart.sh` script (on [the platforms we support](https://learn.netdata.cloud/docs/agent/packaging/platform_support)) and then these repos are used to install native Netdata packages, if possible. If any errors occur, it will fall back to installation of static builds and after that, local builds, in that order.

In my previous post, I said that _you should not use_ `apt install netdata`, as without having the Netdata repos installed via our  `kickstart.sh` script on higher priority, that would install the Netdata packages maintained by the Ubuntu community, **which we do not have control over at all and we do not officially support**.

Long story short, please always use our [Get started with Netdata](https://learn.netdata.cloud/docs/get-started#install-on-linux-with-one-line-installer) guide and let me know if it works for you after full uninstall / re-install.

**Comment by therealgilles** [comment]
Thanks for the explanation. Getting this while trying to do a full uninstall / re-install:

```SHELL
[/root/tmp/netdata-kickstart-BSfVS76vIC]# //usr/libexec/netdata/netdata-updater.sh --enable-auto-updates  
Tue Feb 21 07:00:33 AM PST 2023 : FATAL: netdata-updater.sh: FAILED TO UPDATE NETDATA:  Cannot find environment file or install type file, unable to update.
 FAILED 

 WARNING  Failed to enable auto updates. Netdata will still work, but you will need to update manually.
```

It looks like it's not creating the .environment file when I think it should.

**Comment by therealgilles** [comment]
Side note: The help message specifies `--auto-update-type` as an option but its name is now `--auto-update-method`.

**Comment by therealgilles** [comment]
Only the `--build-only` option seems to generate an `.environment` file.

**Comment by Ferroin** [comment]
@therealgilles Can you share the output of `apt-cache policy netdata`? It sounds like you might be affected by https://github.com/netdata/netdata/issues/12837, and that would help confirm whether this is the case or not.

**Comment by therealgilles** [comment]
I'm getting this:
```SHELL
apt-cache policy netdata
netdata:
  Installed: (none)
  Candidate: 1.38.1
  Version table:
     1.38.1 500
        500 http://repo.netdata.cloud/repos/stable/ubuntu jammy/ Packages
        100 /var/lib/dpkg/status
     1.38.0-116-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-106-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-97-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-90-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-81-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-73-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-71-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-65-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-63-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-51-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-42-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-18-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0-8-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.38.0 500
        500 http://repo.netdata.cloud/repos/stable/ubuntu jammy/ Packages
     1.37.1 500
        500 http://repo.netdata.cloud/repos/stable/ubuntu jammy/ Packages
     1.37.0-260-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-250-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-239-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-231-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-228-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-224-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-199-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-195-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-192-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-182-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0-180-nightly 500
        500 http://repo.netdata.cloud/repos/edge/ubuntu jammy/ Packages
     1.37.0 500
        500 http://repo.netdata.cloud/repos/stable/ubuntu jammy/ Packages
     1.33.1-1ubuntu1 500
        500 http://mirrors.digitalocean.com/ubuntu jammy/universe amd64 Packages
```

**Comment by Ferroin** [comment]
OK, doesn’t look like it’s #12837 after all, though you seem to have both the stable and edge repositories enabled, which is unusual.
